nautobot/users/models.py

```diff
@@ -282,3 +286,73 @@ def list_constraints(self):
         if not isinstance(self.constraints, list):
             return [self.constraints]
         return self.constraints
+
+
+#
+# SavedView
+#
+
+
+class SavedViewMixin(models.Model):
+    """Abstract mixin for enabling Saved View functionality to a given model class."""
+
+    class Meta:
+        abstract = True
+
+    is_saved_view_model = True
+
+
+@extras_features(
+    "custom_validators",
+)
+class SavedView(BaseModel, ChangeLoggedModel):
+    owner = models.ForeignKey(
+        to=User, blank=False, null=False, on_delete=models.CASCADE, help_text="The user that created this view"
+    )
+    name = models.CharField(max_length=CHARFIELD_MAX_LENGTH, blank=False, null=False, help_text="The name of this view")
+    view = models.CharField(
+        max_length=CHARFIELD_MAX_LENGTH,
+        blank=False,
+        null=False,
+        help_text="The name of the list view that the saved view is derived from, e.g. dcim:device_list",
+    )
+    table_config = models.JSONField(
+        encoder=DjangoJSONEncoder, blank=True, default=dict, help_text="Saved Table Config on this view"
+    )
+    pagination_count = models.IntegerField(default=50, blank=True, help_text="Pagination Count of this view")
+    filter_params = models.JSONField(
+        encoder=DjangoJSONEncoder, blank=True, default=dict, help_text="filter parameters that are applied to this view"
+    )
+    sort_order = models.JSONField(
+        encoder=DjangoJSONEncoder,
+        blank=True,
+        default=dict,
+        help_text="table column sort orders that are applied to this view",
+    )
```

Comment:
As I mentioned in standup the other day, I wonder if we're making this model too list-view specific - maybe we should combine all four of these fields into a single JSON field that could store a different dict entirely in the future for customizing say a detail view instead of a list view? Not sure - would that change make the code too much more complex at this time?

Reply:
Collapsed these attributes to one `config` JSONField.
![Screenshot 2024-05-01 at 1 52 36 PM](https://github.com/nautobot/nautobot/assets/46973263/95087a7a-fb99-4b1f-9ca7-7518053bb281)